server/src/main/java/io/crate/execution/ddl/tables/TransportAddColumnAction.java

```diff
@@ -0,0 +1,182 @@
+ */
+
+package io.crate.execution.ddl.tables;
+
+import io.crate.Constants;
+import io.crate.analyze.AnalyzedTableElements;
+import io.crate.common.collections.Maps;
+import io.crate.execution.ddl.AbstractDDLTransportAction;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.PartitionName;
+import io.crate.metadata.cluster.DDLClusterStateHelpers;
+import io.crate.metadata.cluster.DDLClusterStateTaskExecutor;
+import io.crate.metadata.doc.DocTableInfoFactory;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.ClusterStateTaskExecutor;
+import org.elasticsearch.cluster.block.ClusterBlockException;
+import org.elasticsearch.cluster.block.ClusterBlockLevel;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.metadata.IndexTemplateMetadata;
+import org.elasticsearch.cluster.metadata.MappingMetadata;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Singleton;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+
+import static io.crate.metadata.cluster.AlterTableClusterStateExecutor.resolveIndices;
+
+@Singleton
+public class TransportAddColumnAction extends AbstractDDLTransportAction<AddColumnRequest, AcknowledgedResponse> {
+
+    private static final String ACTION_NAME = "internal:crate:sql/table/add_column";
+    private final NodeContext nodeContext;
+
+    @Inject
+    public TransportAddColumnAction(TransportService transportService,
+                                    ClusterService clusterService,
+                                    ThreadPool threadPool,
+                                    NodeContext nodeContext) {
+        super(ACTION_NAME,
+            transportService,
+            clusterService,
+            threadPool,
+            AddColumnRequest::new,
+            AcknowledgedResponse::new,
+            AcknowledgedResponse::new,
+            "add-column");
+        this.nodeContext = nodeContext;
+    }
+
+    @Override
+    public ClusterStateTaskExecutor<AddColumnRequest> clusterStateTaskExecutor(AddColumnRequest request) {
+        return new DDLClusterStateTaskExecutor<>() {
+            @Override
+            protected ClusterState execute(ClusterState currentState, AddColumnRequest request) throws Exception {
+                Metadata.Builder metadataBuilder = Metadata.builder(currentState.metadata());
+                if (request.isPartitioned()) {
```

Comment:
no new test as we already have `testAlterTableAddColumnOnPartitionedTableWithoutPartitions` which covers this case - it used to fail until I did some mixups in this area.